File: vepweb/lib/model/cms/Page.dart
part of vep.model.cms;

class Page {
  int id;
  String canonical;
  String title;
  String content;
  int menu;

  PageForm toPageForm() {
    var pageForm = new PageForm();
    pageForm.canonical = canonical;
    pageForm.title = title;
    pageForm.content = content;
    pageForm.menu = menu;
    return pageForm;
  }
}

class PageForm {
  @jsonIgnore
  String canonical;
  String title;
  String content;
  int menu;
}

File: src/main/scala/vep/app/common/page/Page.scala
package vep.app.common.page

import scalikejdbc.WrappedResultSet
import spray.json.RootJsonFormat
import vep.framework.utils.JsonProtocol

case class Page(
  canonical: String,
  title: String,
  order: Int,
  content: String
)

object Page {
  import JsonProtocol._

  implicit val pageFormat: RootJsonFormat[Page] = jsonFormat4(Page.apply)

  def apply(resultSet: WrappedResultSet): Page = {
    new Page(
      canonical = resultSet.string("canonical"),
      title = resultSet.string("title"),
      order = resultSet.int("navigationOrder"),
      content = resultSet.string("content"),
    )
  }
}